dhall/src/Dhall/Parser/Expression.hs

```diff
@@ -474,14 +474,10 @@ parsers embedded = Parsers {..}
             alternative04 = (do
```

Comment:
The reason I add those `try`s is documented on the source code, but it's because to not loose the `whitespace` consumed.

I'm not to fluent on parser combinators, probably it could be improved

Reply:
Thanks for the explanations!

I think we should do some "left-factoring" here, which IIUC in this case means that we try to parse the first whitespace of a record (type) literal in a central place and then pass it to the following parsers.

@Gabriel439 can probably explain this more clearly.

I think we can address this in a follow-up PR though.

Reply:
Yeah, these `try`s can lead to large parsing performance regressions, so it's worth seeing if we can narrow their scope